bundles/org.openhab.binding.wiz/src/main/java/org/openhab/binding/wiz/internal/handler/WizHandler.java

```diff
@@ -0,0 +1,796 @@
+                logger.trace("Raw packet to send: {}", message);
+
+                // Initialize a datagram packet with data and address
+                DatagramPacket packet = new DatagramPacket(message, message.length, address, DEFAULT_BULB_UDP_PORT);
+
+                // Create a datagram socket, send the packet through it, close it.
+                dsocket = new DatagramSocket(null);
+                dsocket.setReuseAddress(true);
+                dsocket.setBroadcast(true);
+                dsocket.setSoTimeout(500); // Timeout in 500ms
+                dsocket.send(packet);
+                logger.debug("[{}] Sent packet to address: {} and port {}", config.bulbIpAddress, address,
+                        DEFAULT_BULB_UDP_PORT);
+
+                byte[] responseMessage = new byte[1024];
+                packet = new DatagramPacket(responseMessage, responseMessage.length);
+                dsocket.receive(packet);
+
+                return converter.transformResponsePacket(packet);
+            }
+        } catch (SocketTimeoutException e) {
+            logger.trace("[{}] Socket timeout after sending command; no response from {} within 500ms",
+                    config.bulbIpAddress, config.bulbMacAddress);
+        } catch (IOException exception) {
+            logger.debug("[{}] Something wrong happened when sending the packet to port {}... msg: {}",
+                    config.bulbIpAddress, DEFAULT_BULB_UDP_PORT, exception.getMessage());
+        } finally {
+            if (dsocket != null) {
+                dsocket.close();
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Sends a setPilot request and checks for success
+     */
+    private synchronized boolean setPilotCommand(final @Nullable Param param) {
+        WizResponse response = sendRequestPacket(WizMethodType.SetPilot, param);
+        if (response != null) {
+            boolean setSucceeded = response.getResultSuccess();
+            if (setSucceeded) {
+                // can't process this response it doens't have a syncstate, so request updated state
+                // let the getPilot response update the timestamps
+                try {
+                    // wait for state change to apply
+                    Thread.sleep(1000L);
+                } catch (InterruptedException e) {
+                }
+                getPilot();
+                return setSucceeded;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Makes note of the latest timestamps and sets the bulb online
+     */
+    private synchronized void updateTimestamps() {
+        if (hasConfigurationError() || disposed) {
+            return;
+        }
+        updateStatus(ThingStatus.ONLINE);
+        latestUpdate = System.currentTimeMillis();
+        latestOfflineRefresh = System.currentTimeMillis();
+        updateState(CHANNEL_LAST_UPDATE, new DateTimeType());
```

Comment:
The empty constructor uses system time-zone rather than the time-zone configured in openHAB. `TimeZoneProvider` would be needed to update with the respect to this time-zone.